tests/mock_crust/utils.rs

```diff
@@ -381,15 +381,17 @@ pub fn poll_all_until(
 /// Polls and processes all events, until there are no unacknowledged messages left.
 pub fn poll_and_resend(nodes: &mut [TestNode], clients: &mut [TestClient]) {
     let dummy = |_nodes: &[TestNode]| false;
-    poll_and_resend_until(nodes, clients, &dummy)
+    poll_and_resend_until(nodes, clients, &dummy, None)
 }
 
 /// Polls and processes all events, until there are no unacknowledged messages left.
 /// should_stop: can be used for an early return from poll_and_resend
+/// extra_advance: this is so far only used for the ignoring candidate_info test.
 pub fn poll_and_resend_until(
     nodes: &mut [TestNode],
     clients: &mut [TestClient],
     should_stop: &Fn(&[TestNode]) -> bool,
+    mut extra_advance: Option<u64>,
```

Comment:
Why do we need that field? which test fail if we do it unconditionally?

Reply:
Lets keep it as you did for now, we can reconsider how best to improve it later.

Reply:
>Why do we need that field
that's for the `ignoring_candidate_info` test.
>which test fail if we do it unconditionally
the new `joining_node_with_ignoring_candidate_info` test